cni/pkg/ambient/server.go

```diff
@@ -50,6 +50,9 @@ type Server struct {
 	iptablesCommand lazy.Lazy[string]
 	redirectMode    RedirectMode
 	ebpfServer      *ebpf.RedirectServer
+
+	// podReconcileHandler is implemented by Server, but it can be overridden by tests.
+	podReconcileHandler podReconcileHandler
```

Comment:
I like this approach - I would prefer to split the real handler impl out from `Server` fully since `Server` is already too big IMO and provide it to Server as a constructor arg rather than relying on private overrides - but this is an improvement, ty!

Reply:
If separate this handler out from Server, this handler needs more dependencies, I don’t want to do much refactoring on it. I think this is enough for unit test. I want to leave it to next time, we still need to test the namespace reconcile.
Anyway, fixed other comments